iree_tests/conftest.py

```diff
@@ -280,13 +283,25 @@ def collect(self):
             #   try pytest-dependency or pytest-depends
             for test_case in test_cases:
                 test_name = config_name + "_" + test_case.name
+                # use backend specific runtime flagfile if it exists (don't just add because order matters)
+                config_specific_flagfile = test_directory / (test_case.runtime_flagfile.split(".")[0] + "_" + config_name + ".txt")
```

Comment:
This requires that downstream config files (e.g. https://github.com/iree-org/iree/blob/main/build_tools/pkgci/external_test_suite/pytorch_models_cpu_llvm_task.json) use names matching the upstream flagfiles (e.g. `iree_tests/pytorch/models/sdxl-scheduled-unet-3-tank/real_weights_data_flags_cpu_llvm_task.txt`). That will make it harder to add new configurations -- especially with out of tree targets like https://github.com/nod-ai/iree-amd-aie . Downstream should be able to just reference the files provided by the test suite, it should not need to make changes to the test suite itself.

A few ideas:

## One config per model (when needed)

Add a new .json config file for each set of flags.

https://github.com/iree-org/iree/blob/main/build_tools/pkgci/external_test_suite/pytorch_models_cpu_llvm_task.json could be forked into `pytorch_model_sdxl_scheduled_unet_cpu_llvm_task.json`, with extra flags added as needed and then tested with `pytest SHARK-TestSuite/iree_tests/pytorch/models/sdxl-scheduled-unet`

With issues like https://github.com/iree-org/iree/issues/17137, I'm expecting we'll have a few configs per model/platform that we'll want to test, like "default flags", "performance flags (config A)", "performance flags (config b)".

## Redefine the config format

The config files are JSON at the moment to keep them simple. We can replace them with something more powerful like Python to add whatever logic we want. For example, see how test cases are defined in https://github.com/iree-org/iree/blob/main/experimental/regression_suite/tests/pregenerated/test_llama2.py . The ONNX test suite has 1000+ test cases, so I wanted to lean on something more automated like the JSON configs for it. For model tests, being explicit about each test case might be easier.

Reply:
For redefining the config format, I believe we can write Python code that substitutes for the "collect" phase in pytest, replacing this: https://github.com/nod-ai/SHARK-TestSuite/blob/2dba6a8e4d56ca146e25077cfa96f82001d2bc33/iree_tests/conftest.py#L248-L296

Could try using `IreeCompileAndRunTestSpec` directly, or extract some parts of that file into fixtures (like https://github.com/iree-org/iree/blob/main/experimental/regression_suite/ireers/fixtures.py, which we could also just use directly) or other helper functions.

I think we can keep these parts of testing infrastructure pretty separable:
* Model files, parameters, inputs, outputs
* Automated test case discovery (can replace with explicit/manual test definitions)
* Device/configuration-specific flags
* Test running

Reply:
For this PR I'd probably start with one config per model though. (work with what we have for N=1 and N=2, maybe up to N=10... generalize and refactor to reduce duplication and make maintenance easier when N=3+).

Reply:
Ok yeah this all makes sense. Will create config files where needed for now.